src/core/tests/type_prop/arithmetic_ops.hpp

```diff
@@ -172,19 +345,300 @@ TYPED_TEST_P(ArithmeticOperator, dynamic_shape_5D) {
+        Dimension dim_0_A = Dimension(2, 4);
+        Dimension dim_0_B = Dimension(-1);
+
+        ov::DimensionTracker::set_label(dim_0_A, 10);
+        ov::DimensionTracker::set_label(dim_0_B, 20);
+
+        PartialShape pshape_A = {dim_0_A, 3, 224, 1}, pshape_B = {dim_0_B, 3, 1, 224};
+
+        auto param_A = std::make_shared<op::Parameter>(element::f32, pshape_A);
+        auto param_B = std::make_shared<op::Parameter>(element::f32, pshape_B);
+        const auto op = std::make_shared<TypeParam>(param_A, param_B);
+
+        const auto out_shape = op->get_output_partial_shape(0);
+        PartialShape expected_shape = {Dimension(2, 4), 3, 224, 224};
+
+        EXPECT_EQ(out_shape, expected_shape);
+        EXPECT_EQ(ov::DimensionTracker::get_label(out_shape[0]), 10);
+        EXPECT_EQ(ov::DimensionTracker::get_label(out_shape[1]), 0);
+        EXPECT_EQ(ov::DimensionTracker::get_label(out_shape[2]), 0);
+        EXPECT_EQ(ov::DimensionTracker::get_label(out_shape[3]), 0);
+    }
+    {  // Both params have dynamic interval dimension labels
+        Dimension dim_0_A = Dimension(2, 4);
+        Dimension dim_0_B = Dimension(2, 4);
+
+        ov::DimensionTracker::set_label(dim_0_A, 10);
+        ov::DimensionTracker::set_label(dim_0_B, 20);
+
+        PartialShape pshape_A = {dim_0_A, 3, 224, 1}, pshape_B = {dim_0_B, 3, 1, 224};
+
+        auto param_A = std::make_shared<op::Parameter>(element::f32, pshape_A);
+        auto param_B = std::make_shared<op::Parameter>(element::f32, pshape_B);
+        const auto op = std::make_shared<TypeParam>(param_A, param_B);
+
+        const auto out_shape = op->get_output_partial_shape(0);
+        PartialShape expected_shape = {Dimension(2, 4), 3, 224, 224};
+
+        EXPECT_EQ(out_shape, expected_shape);
+        EXPECT_EQ(ov::DimensionTracker::get_label(out_shape[0]), 0);  // TODO: Shouldn't be 10 or 20?
+        EXPECT_EQ(ov::DimensionTracker::get_label(out_shape[1]), 0);
+        EXPECT_EQ(ov::DimensionTracker::get_label(out_shape[2]), 0);
+        EXPECT_EQ(ov::DimensionTracker::get_label(out_shape[3]), 0);
+    }
+    {  // Both params have fully dynamic dimension and labels
+        Dimension dim_0_A = Dimension(-1);
+        Dimension dim_0_B = Dimension(-1);
+
+        ov::DimensionTracker::set_label(dim_0_A, 10);
+        ov::DimensionTracker::set_label(dim_0_B, 20);
+
+        PartialShape pshape_A = {dim_0_A, 3, 224, 1}, pshape_B = {dim_0_B, 3, 1, 224};
+
+        auto param_A = std::make_shared<op::Parameter>(element::f32, pshape_A);
+        auto param_B = std::make_shared<op::Parameter>(element::f32, pshape_B);
+        const auto op = std::make_shared<TypeParam>(param_A, param_B);
+
+        const auto out_shape = op->get_output_partial_shape(0);
+        PartialShape expected_shape = {-1, 3, 224, 224};
+
+        EXPECT_EQ(out_shape, expected_shape);
+        EXPECT_EQ(ov::DimensionTracker::get_label(out_shape[0]), 0);  // TODO: Shouldn't be 10 or 20?
```

Comment:
In case there are two fully dynamic dimensions (without bounds) with different label, they can be not equal for example `1` and `128`, so label should be reset to `0`.